src/lib/journey/stages/generic.svelte

```diff
@@ -84,12 +86,12 @@
   needsFocus={formNeedsFocus}
   onSubmitWhenValid={submitFormWrapper}
 >
-  {#if displayIcon}
+  {#if form?.icon}
     <div class="tw_flex tw_justify-center">
       <ShieldIcon classes="tw_text-gray-400 tw_fill-current" size="72px" />
     </div>
   {/if}
-  <header id={formHeaderId}>
+  <header bind:this={linkWrapper} id={formHeaderId}>
```

Comment:
This `linkWrapper` is what captures the specific link behavior and handles them internally if they match one of the configured journeys.